### PR TITLE
Register Site Editor Templates.

### PR DESCRIPTION
**Pull request by EugeneKyale**
## Ticket
[TEC-4790](https://theeventscalendar.atlassian.net/browse/TEC-4790)

## 🎯Objective
- Introduced registration for both `Events Archive` and `Single Event` templates.
- Implemented logic to toggle the `Activate Block Editor for Events` setting based on the full site editor theme status.
- Modified the `Events → Display → Events Template` setting, ensuring it defaults to Default Events Template when a site editor theme is activated.

## 🎥 Screencast
https://drive.google.com/file/d/1c6QOprxT8bG9pH7kRs6tl46Sl6eEKNtS/view?usp=sharing

## 📸 Screenshot
site_editor_templates

[TEC-4790]: https://theeventscalendar.atlassian.net/browse/TEC-4790?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by stratease** [comment]
Perhaps we need some snapshot tests at least?

File: src/Events/Editor/Full_Site/Templates.php [hunk 1]
```diff
@@ -26,14 +26,23 @@ class Templates {
...
+
 	/**
 	 * Modify the available Templates so that people can edit the template.
 	 *
 	 * @since 5.14.2
 	 *
 	 * @param WP_Block_Template[] $query_result Array of found block templates.
-	 * @param array  $query {
-	 *     Optional. Arguments to retrieve templates.
+	 * @param array               $query        {
```
Comment (stratease) [inline src/Events/Editor/Full_Site/Templates.php:44]:
```suggestion
	 * @param array               $query        
```
I know this wasn't you, but is that curly brace intended? Looks off to me.

File: src/Events/Editor/Full_Site/Templates.php [hunk 2]
```diff
@@ -82,9 +87,71 @@ public function get_template_events_archive() {
 		$template->content        = Template_Utils::inject_theme_attribute_in_content( $template_content );
 		$template->slug           = static::$archive_slug;
 		$template->source         = 'custom';
+		$template->type           = 'wp_template';
+		$template->title          = esc_html__( 'Calendar Views (Event Archive)', 'the-events-calendar' );
```
Comment (stratease) [inline src/Events/Editor/Full_Site/Templates.php:91]:
```suggestion
		$template->title          = esc_html_x( 'Calendar Views (Event Archive)', 'The Full Site editor block navigation title' 'the-events-calendar' );
```
Perhaps something like this? I think we want to use context variations most of the time now right?

File: src/Events/Editor/Full_Site/Templates.php [hunk 3]
```diff
@@ -82,9 +87,71 @@ public function get_template_events_archive() {
...
+
+		$template                 = new WP_Block_Template();
+		$template->id             = 'the-events-calendar//single-event';
 		$template->theme          = 'The Events Calendar';
+		$template->content        = Template_Utils::inject_theme_attribute_in_content( $template_content );
+		$template->slug           = static::$single_slug;
+		$template->source         = 'custom';
 		$template->type           = 'wp_template';
-		$template->title          = esc_html__( 'Events Archive', 'the-events-calendar' );
+		$template->title          = esc_html__( 'Event Single', 'the-events-calendar' );
```
Comment (stratease) [inline src/Events/Editor/Full_Site/Templates.php:153]:
```suggestion
		$template->title          = esc_html_x( 'Event Single', 'Context here...', 'the-events-calendar' );
```
etc

File: src/Tribe/Editor/Compatibility.php [hunk 4]
```diff
@@ -238,4 +239,25 @@ public function filter_is_classic_editor( $is_classic_editor = false ) {
...
+	 * @param string   $old_name  The name of the old theme that was active before switching.
+	 * @param WP_Theme $old_theme WP_Theme instance of the old theme.
+     *
+     * @return void
+     */
+    public function toggle_blocks_editor_setting_on_theme_switch( $old_name, $old_theme = false ) {
+        if ( tec_is_full_site_editor() ) {
+            tribe_update_option( static::$blocks_editor_key, true );
+        } else {
+            tribe_update_option( static::$blocks_editor_key, false );
```
Comment (stratease) [inline src/Tribe/Editor/Compatibility.php:260]:
```suggestion
            tribe_update_option( static::$blocks_editor_key, false );
```
Do we really need to disable it? Perhaps it was always set. Just a thought for simplicity / maintenance sake.

**Review by stratease — CHANGES_REQUESTED** [review]

File: src/Events/Editor/Full_Site/Templates/single-event.html [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<!-- wp:template-part {"slug":"header","tagName":"header"} /-->
+<!-- wp:tribe/single-event {} /-->
```
Comment (stratease) [inline src/Events/Editor/Full_Site/Templates/single-event.html:2]:
```
<!-- wp:tribe/single-event {} /-->
```
More of a question here.

Is the `tribe/single-event` a reference to our other block? I assume we will not be using the existing blocks for this. 

Also wondering if we should name it like `tec-events/...` or what our naming convention might be (vs `tribe`)?

**Comment by EugeneKyale** [comment]
> Optional. Arguments to retrieve templates.

Yeah I'll a snapshot test for this.

File: src/Events/Editor/Full_Site/Templates.php (hunk 1 above)
Comment (EugeneKyale) [reply src/Events/Editor/Full_Site/Templates.php:44]: I think that's intentional. Additional details provided in the curly braces give a more in-depth description of the structure and intent of this $query parameter.

File: src/Tribe/Editor/Compatibility.php (hunk 4 above)
Comment (EugeneKyale) [reply src/Tribe/Editor/Compatibility.php:260]: Ok yeah it makes sense to only update the option on FSE theme activation. I've changed it and updated the docblock to reflect the update.

File: src/Events/Editor/Full_Site/Templates.php (hunk 2 above)
Comment (EugeneKyale) [reply src/Events/Editor/Full_Site/Templates.php]: Updates

File: src/Events/Editor/Full_Site/Templates.php (hunk 3 above)
Comment (EugeneKyale) [reply src/Events/Editor/Full_Site/Templates.php]: Updated

File: src/Events/Editor/Full_Site/Templates/single-event.html (hunk 5 above)
Comment (bordoni) [reply src/Events/Editor/Full_Site/Templates/single-event.html:2]: `tribe/` is the old namespace for blocks, we will need at some point to create a new one and have the old ones map to this new one.

**Pull request merged**